src/views/Home/components/Carousel/index.js

```diff
@@ -1,76 +1,142 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 
 import cms from 'services/cms';
 import ImageGallery from 'react-image-gallery';
 import GordonLoader from 'components/Loader';
 
-export default class GordonCarousel extends Component {
-  constructor(props) {
-    super(props);
+const GordonCarousel = () => {
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [carouselContent, setCarouselContent] = useState({});
+  const [imageGallery, setImageGallery] = useState(null)
 
-    this.loadCarousel = this.loadCarousel.bind(this);
-
-    this.state = {
-      error: null,
-      loading: true,
-      carouselContent: {},
+  useEffect(() => {
+    const loadCarousel = async () => {
+      setLoading(true);
+      try {
+        setCarouselContent(await cms.getSlides());
+        setLoading(false);
+      } catch (error) {
+        setError(error);
+      }
     };
+    loadCarousel();
+  }, []);
 
-    this.linkArray = [];
-  }
-  componentDidMount() {
-    this.loadCarousel();
-  }
-  async loadCarousel() {
-    this.setState({ loading: true });
-    try {
-      const carouselContent = await cms.getSlides();
-      this.setState({ loading: false, carouselContent });
-      this.state.carouselContent.map((slide) => this.linkArray.push(slide.ActionLink));
-    } catch (error) {
-      this.setState({ error });
-    }
-  }
+  const handleClickSlide = useCallback(
+    () => {
+      if (carouselContent[imageGallery.getCurrentIndex()].ActionLink !== '') {
+        window.location = carouselContent[imageGallery.getCurrentIndex()].ActionLink;
+      }
+    },
+    [carouselContent, imageGallery],
+  );
 
-  handleClickSlide() {
-    if (this.state.carouselContent[this._imageGallery.getCurrentIndex()].ActionLink !== '') {
-      window.location = this.state.carouselContent[this._imageGallery.getCurrentIndex()].ActionLink;
-    }
+  if (error) {
+    throw error;
```

Comment:
As above, if our only way of handling the error is to rethrow it, it's better to not catch it at all.